.github/workflows/build.yml

```diff
@@ -648,7 +648,70 @@ jobs:
     ####################################################
     # Test Dependencies/Regressions                    #
     ####################################################
-    test_dependencies:
+    test_buildtime_dependencies:
```

Comment:
will we still have tests running with numpy < 2 for a while?
Main thing I want to make sure is tested is running in a numpy < 2 env with the numpy >=2 build

Reply:
this PR doesn't change how csp builds / deploys, it just adds an extra step to build a wheel against numpy 2 and test it. This wheel doesnt get deployed yet.

The PR description contains more info, we can move to build everything against numpy 2 if we drop python 3.8, alternatively we can build 3.8 against numpy<2 and 3.9+ against numpy>2, which is slightly less preferrable imo